docs/runtime/core.md

```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
 Runtime Core has four modules. These are namespaces as well as directory names in `/runtime/onert/core/src/`.
 
 - `ir`  stands for Intermediate Representation which contains Neural Network Graph data structures
-- `compiler` converts IR to executable format
-- `exec` is execution module which is the result of compilation
-- `backend` is backend interface
+- `compiler` converts IR to an executable format
+- `exec` is an execution module which is the result of a compilation
+- `backend` is a backend interface
```

Comment:
seems like a recursive definition. :-)
I guess  we need some explanation of _backend_. (People who have experience with server may think `database` when they hear `backend`. :-P)

 I found this in this doc. How about using it here?
```
Backend interface classes are mostly about memory management and kernel generation.
```
or something like this?

_backend is provides API and implementation related to real calculation of operation. Some of its roles are providing execution logic optimzied for a specific hardware such as CPU, GPU or NPU, and memory management._

Reply:
This is a very short summary for so I would like to keep it as a one-liner. I may add a bit more words for that later. (I won't make change for that in this PR as it is not relevant.)